actioncable/lib/assets/javascripts/action_cable/connection.coffee

```diff
@@ -75,7 +79,17 @@ class ActionCable.Connection
     error: ->
       @disconnect()
 
+    dispatchEvent: (identifier, type, message) ->
+      unless identifier == "_ping"
+        event = document.createEvent('Event')
+        event.initEvent("cable:#{type}", true, false)
+        channel = JSON.parse(identifier).channel
+        event.cable = {data: message, identifier: channel}
+        document.dispatchEvent(event)
+
   disconnect: ->
     return if @disconnected
     @disconnected = true
-    @consumer.subscriptions.notifyAll("disconnected")
+    notification_type = "disconnected"
+    @consumer.subscriptions.notifyAll(notification_type)
+    @events.dispatchEvent("{\"channel\": \"All\"}", notification_type)
```

Comment:
Using the above suggestion this would be:

```coffee
@events.dispatchEvent("disconnected")
```

This avoids the raw JSON and makes the API nicer, because people can assume that `channel == null` means all channels. No need for a special key.